### PR TITLE
[ETFE-2632] added ItemBulkPackagingChoice page

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: test/controllers/BaseNavigationControllerSpec.scala [hunk 1]
```diff
@@ -231,7 +231,7 @@ class BaseNavigationControllerSpec extends SpecBase with GuiceOneAppPerSuite wit
     }
   }
 
-  "validateIndex" - {
+  "validateIndexAsync" - {
```
Comment (LewisMitchell) [inline test/controllers/BaseNavigationControllerSpec.scala:234]: Should this be done in the BaseItemNavControllerSpec instead? This remains unchanged?

File: test/views/sections/items/ItemCommodityCodeViewSpec.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ class ItemCommodityCodeViewSpec extends ViewSpecBase with ViewBehaviours {
       val view = app.injector.instanceOf[ItemCommodityCodeView]
       val form = app.injector.instanceOf[ItemCommodityCodeFormProvider].apply()
       val submitRoute = controllers.sections.items.routes.ItemCommodityCodeController.onPageLoad(testErn, testDraftId, testIndex1, NormalMode)
-      val testEpc = "testEpc"
+      val testEpc = "testEpcTobacco"
```
Comment (LewisMitchell) [inline test/views/sections/items/ItemCommodityCodeViewSpec.scala:50]: Does this need to change?
```suggestion
      val testEpc = testEpcTobacco
```

File: test/views/sections/items/ItemCommodityCodeViewSpec.scala [hunk 3]
```diff
@@ -58,7 +58,7 @@ class ItemCommodityCodeViewSpec extends ViewSpecBase with ViewBehaviours {
       )
 
       val testCommodityCode2 = CnCodeInformation(
-        cnCode = "testCnCode2",
+        cnCode = "testCnCodeWine",
```
Comment (LewisMitchell) [inline test/views/sections/items/ItemCommodityCodeViewSpec.scala:61]: Does this need to change?
```suggestion
        cnCode = testCnCodeWine,
```

File: test/controllers/sections/items/ItemBulkPackagingChoiceControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    "must redirect to the Index page for POST when no Goods Type" in new Test(Some(
+      emptyUserAnswers.set(ItemCommodityCodePage(testIndex1), "12345768")
+    )) {
+      val result: Future[Result] = controller.onSubmit(testErn, testDraftId, testIndex1, NormalMode)(FakeRequest().withFormUrlEncodedBody(("value", "true")))
+
+      status(result) mustEqual SEE_OTHER
+      redirectLocation(result).value mustEqual routes.ItemsIndexController.onPageLoad(testErn, testDraftId).url
+    }
+
+    "must redirect to the Index page for GET when no Item" in new Test(Some(emptyUserAnswers)) {
```
Comment (LewisMitchell) [inline test/controllers/sections/items/ItemBulkPackagingChoiceControllerSpec.scala:123]: Is this testing where no EPC is in the user answers?

File: test/controllers/sections/items/ItemBulkPackagingChoiceControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      val result: Future[Result] = controller.onPageLoad(testErn, testDraftId, testIndex1, NormalMode)(FakeRequest())
+
+      status(result) mustEqual SEE_OTHER
+      redirectLocation(result).value mustEqual routes.ItemsIndexController.onPageLoad(testErn, testDraftId).url
+    }
+
+    "must redirect to the Index page for POST when no Item" in new Test(Some(emptyUserAnswers)) {
```
Comment (LewisMitchell) [inline test/controllers/sections/items/ItemBulkPackagingChoiceControllerSpec.scala:130]: Is this testing where no EPC is in the user answers?

File: test-utils/fixtures/messages/sections/items/ItemBulkPackagingChoiceMessages.scala [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  sealed trait ViewMessages extends BaseMessages {
+    _: i18n =>
+    val heading: String => String = goodsType => s"Is the $goodsType being moved in bulk?"
+    val title: String => String = goodsType => titleHelper(heading(goodsType))
+    val caption = "This section is Item information"
+    val hint = "Moving goods in bulk means that the goods are loaded directly into a vessel in large quantity, and not packaged. For example, transporting liquids in a tanker."
+    val cyaLabel: String = "Bulk packaging"
+    val cyaChangeHidden: String = "bulk packaging"
+    val errorRequired: String => String = goodsType => s"Select yes if the $goodsType is being moved in bulk"
```
Comment (LewisMitchell) [inline test-utils/fixtures/messages/sections/items/ItemBulkPackagingChoiceMessages.scala:31]: This isn't being used, should it be removed?

File: test/navigation/ItemsNavigatorSpec.scala [hunk 7]
```diff
@@ -360,6 +360,42 @@ class ItemsNavigatorSpec extends SpecBase {
         }
 
       }
+
+      "must go from the ItemBulkPackagingChoicePage" - {
```
Comment (LewisMitchell) [inline test/navigation/ItemsNavigatorSpec.scala:364]: I think a test is missing for CAM-ITM24

File: test/views/sections/items/ItemBulkPackagingChoiceViewSpec.scala [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+              val view = app.injector.instanceOf[ItemBulkPackagingChoiceView]
+              val form = app.injector.instanceOf[ItemBulkPackagingChoiceFormProvider].apply(goodsType)
+
+              implicit val doc: Document = Jsoup.parse(view(form, testOnwardRoute, goodsType).toString())
+
+              behave like pageWithExpectedElementsAndMessages(Seq(
+                Selectors.title -> messagesForLanguage.title(goodsType.toSingularOutput()),
+                Selectors.h1 -> messagesForLanguage.heading(goodsType.toSingularOutput()),
+                Selectors.h2(1) -> messagesForLanguage.caption,
```
Comment (LewisMitchell) [inline test/views/sections/items/ItemBulkPackagingChoiceViewSpec.scala:58]: Is it worth adding a test for the caption?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: conf/messages [hunk 9]
```diff
@@ -1075,3 +1075,10 @@ itemFiscalMarksChoice.heading = Does the {0} have any fiscal marks?
 itemFiscalMarksChoice.checkYourAnswersLabel = Fiscal marks
 itemFiscalMarksChoice.error.required = Select yes if the {0} has fiscal marks
 itemFiscalMarksChoice.change.hidden = fiscal marks
+
+itemBulkPackagingChoice.title = Is the {0} being moved in bulk?
+itemBulkPackagingChoice.heading = Is the {0} being moved in bulk?
+itemBulkPackagingChoice.hint = Moving goods in bulk means that the goods are loaded directly into a vessel in large quantity, and not packaged. For example, transporting liquids in a tanker.
+itemBulkPackagingChoice.checkYourAnswersLabel = Bulk packaging
+itemBulkPackagingChoice.error.required = Select yes if the {0} are being moved in bulk
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply conf/messages:1083]: I raised that as a problem with UCD, got the copydeck changed and then forgot to change the code 🫠 good spot

File: test/navigation/ItemsNavigatorSpec.scala [hunk 10]
```diff
@@ -356,7 +355,7 @@ class ItemsNavigatorSpec extends SpecBase {
           }
         }
 
-        "to the Items index page" - {
+        "to the Packaging Select page" - {
```
Comment (LewisMitchell) [inline test/navigation/ItemsNavigatorSpec.scala:358]: Needs to change back

File: test/navigation/ItemsNavigatorSpec.scala [hunk 11]
```diff
@@ -403,6 +402,41 @@ class ItemsNavigatorSpec extends SpecBase {
...
+          //TODO: Redirect to CAM-ITM15
+          "when the user answers no and EPC is wine" in {
+            navigator.nextPage(ItemBulkPackagingChoicePage(testIndex1), NormalMode, emptyUserAnswers
+              .set(ItemExciseProductCodePage(testIndex1), "W200")
+              .set(ItemBulkPackagingChoicePage(testIndex1), false)
+            ) mustBe testOnly.controllers.routes.UnderConstructionController.onPageLoad()
+          }
+        }
+
+        "to the Items Index page" - {
```
Comment (LewisMitchell) [inline test/navigation/ItemsNavigatorSpec.scala:428]: Needs to change

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**